lib/rules/lines-around-comment.js

```diff
@@ -207,12 +202,7 @@ module.exports = {
          * @returns {boolean} True if the comment is at parent end.
          */
         function isCommentAtParentEnd(node, nodeType) {
-            const ancestors = context.getAncestors();
-            let parent;
-
-            if (ancestors.length) {
-                parent = ancestors.pop();
-            }
+            const parent = sourceCode.getNodeByRangeIndex(node.range[0]);
```

Comment:
I ended up extracting it into its own method as well as renamed the incorrect `node` parameters to `token`. Hopefully this makes it all clearer, but Let me know if it doesn't :)